src/test/rgw/test_d4n_filter.cc

```diff
@@ -0,0 +1,2810 @@
+    EXPECT_EQ(fs::exists(CACHE_DIR + "/" + TEST_BUCKET + testName + "/" + TEST_OBJ + testName + "/" + oid, err), true);     
+    oid = instance + "#0#" + std::to_string(ofs);
+    EXPECT_EQ(fs::exists(CACHE_DIR + "/" + TEST_BUCKET + testName + "/" + TEST_OBJ + testName + "/" + instance, err), true);     
+
+    // For version enabled object
+    std::unique_ptr<rgw::sal::Object::DeleteOp> del_op_enabled = objEnabled->get_delete_op();
+    EXPECT_EQ(del_op_enabled->delete_obj(env->dpp, optional_yield{yield}, rgw::sal::FLAG_LOG_OP), 0);
+
+    // For version suspended object
+    std::unique_ptr<rgw::sal::Object::DeleteOp> del_op_suspended = objSuspended->get_delete_op();
+    EXPECT_EQ(del_op_suspended->delete_obj(env->dpp, optional_yield{yield}, rgw::sal::FLAG_LOG_OP), 0);
+
+    {
+      boost::system::error_code ec;
+      request req;
+      req.push("EXISTS", TEST_BUCKET + testName + "_" + TEST_OBJ + testName + "_0_0");
+      req.push("EXISTS", TEST_BUCKET + testName + "_" + TEST_OBJ + testName + "_0_9");
+      req.push("EXISTS", TEST_BUCKET + testName + "__:" + instance + "_" + TEST_OBJ + testName + "_0_0");
+      req.push("EXISTS", TEST_BUCKET + testName + "__:" + instance + "_" + TEST_OBJ + testName + "_0_9");
+      req.push("EXISTS", TEST_BUCKET + testName + "__:null_" + TEST_OBJ + testName + "_0_0");
+      req.push("FLUSHALL");
+
+      response< int, int, int, int, int, boost::redis::ignore_t > resp;
+
+      conn->async_exec(req, resp, yield[ec]);
+      ASSERT_EQ((bool)ec, false);
+      EXPECT_EQ((int)std::get<0>(resp).value(), 0);
+      EXPECT_EQ((int)std::get<1>(resp).value(), 0);
+      EXPECT_EQ((int)std::get<2>(resp).value(), 0);
+      EXPECT_EQ((int)std::get<3>(resp).value(), 0);
+      EXPECT_EQ((int)std::get<4>(resp).value(), 0);
+    }
+
+    /* TODO: Eviction cycle to delete cache blocks
+    EXPECT_EQ(fs::exists(CACHE_DIR + "/" + TEST_BUCKET + testName + "/" + TEST_OBJ + testName + "/" + instance, err), false);  
+    EXPECT_EQ(fs::exists(CACHE_DIR + "/" + TEST_BUCKET + testName + "/" + TEST_OBJ + testName + "/" + oid, err), false);     
+    EXPECT_EQ(fs::exists(CACHE_DIR + "/" + TEST_BUCKET + testName + "/" + TEST_OBJ + testName + "/" + version, err), false);  
+    oid = version + "#0#" + std::to_string(ofs);
+    EXPECT_EQ(fs::exists(CACHE_DIR + "/" + TEST_BUCKET + testName + "/" + TEST_OBJ + testName + "/" + oid, err), false);     
+    */
+
+    conn->cancel();
+    testBucket->remove(env->dpp, true, optional_yield{yield});
+    DriverDestructor driver_destructor(static_cast<rgw::sal::D4NFilterDriver*>(driver));
+  }, rethrow);
+
+  io.run();
+}
+
+// Write cache tests, unversioned
+TEST_F(D4NFilterFixture, PutObjectWrite)
+{
+  env->cct->_conf->d4n_writecache_enabled = true;
+  env->cct->_conf->rgw_d4n_cache_cleaning_interval = 0.5;
+  const std::string testName = "PutObjectWrite";
+  const std::string bucketName = "/tmp/d4n_filter_tests/dbstore-default_ns.1";
```

Comment:
can you explain here why you have prefixed the bucket name with the directory path?

Reply:
I did this because the filter sets dirty entries with the bucket id rather than the bucket name during several calls, so when the cleaning method executes, it searches for the DBStore bucket path.